lib/md043.js

```diff
@@ -21,20 +22,21 @@ module.exports = {
       let hasError = false;
       let anyHeadings = false;
       const getExpected = () => requiredHeadings[i++] || "[None]";
+      const handleCase = (str) => (matchCase ? str : str.toLowerCase());
       forEachHeading(params, (heading, content) => {
         if (!hasError) {
           anyHeadings = true;
-          const actual = levels[heading.tag] + " " + content;
-          const expected = getExpected();
+          const actual = levels[heading.tag] + " " + handleCase(content);
```

Comment:
Based on the bug I call out in the output, it may be necessary to leave actual and expected alone and apply handleCase() in the comparisons below so actual and expected have the original case when passed to addErrorDetailIf.